### PR TITLE
Convert poll sampler into sections

### PR DESCRIPTION
**Pull request by dfabulich**
Now we have two separate sections, "New Polls" and "Polls with Recent Votes". (The "Browse Polls" link now links to a sort by most votes.)

### CODE REVIEW

**Comment by alice-blue** [comment]
poll sections

* The links seem to work correctly.
* As you can see in the screenshot, the first poll on both lists is the same poll. I think that if a poll appears in the top section, it should not be repeated in the bottom section.
* The polls section is now longer (taller?) than before--it won't all fit on my screen at once, and the "New to IF?" section is still pretty short. (I forget what the plan was for that section.)  The difference looks a little odd, but not a huge deal.
* It might be clearer to say "polls with recent votes" vs. just a heading that says "recent votes."

**Comment by dfabulich** [comment]
> I think that if a poll appears in the top section, it should not be repeated in the bottom section.

That's possible, but thorny. If by chance all of the polls with recent votes are new, would we then hide the section? (And its "See More" link?) If we hide the section, would we hide both section headers? (Hide both "See More" links?)

Or what if there's exactly one "Recent Votes" poll that doesn't appear in the "New Polls" section?

(I suspect that this line of reasoning is part of why the old sampler algorithm just mushed them all together in one unlabeled section.)

> the "New to IF?" section is still pretty short. (I forget what the plan was for that section.)

My current plan is https://github.com/iftechfoundation/ifdb/issues/415, running an annual poll for "games for beginners." I'm hoping we can run that poll at the start of 2025.

**Comment by dfabulich** [comment]
> It might be clearer to say "polls with recent votes" vs. just a heading that says "recent votes."

Done

File: www/components/poll-sampler.php [hunk 1]
```diff
@@ -25,115 +25,58 @@
...
+    . "<p>Help other IFDB members find the games they're looking for "
+    . "by voting in their polls.  Here are a few recent ones:</p>";
+
+global $nonce;
+echo "<style nonce='$nonce'>\n"
+    . ".poll-sampler__create { margin-top: 1ex; }\n"
+    . "</style>\n";
+
+function displayPoll($row) {
+    [$pollID, $pollTitle, $pollDesc, $pollDate, $pollUserID,
```
Comment (salty-horse) [inline www/components/poll-sampler.php]: unused fields that can be removed from the select:
`$pollDesc`, `$pollDate`, `$pollVotes`, `$pollGames`, `$pollLastVoteDate`, `$pollUserName`.

**Review by salty-horse — COMMENTED** [review]
Duplicates-removal is possible, since that's what the old code did.

It's mostly a matter of fetching more than needed, and then deciding which list should get priority if there's a duplicate.

**Comment by alice-blue** [comment]
> > I think that if a poll appears in the top section, it should not be repeated in the bottom section.
> 
> That's possible, but thorny. If by chance all of the polls with recent votes are new, would we then hide the section? (And its "See More" link?) If we hide the section, would we hide both section headers? (Hide both "See More" links?)
> 
> Or what if there's exactly one "Recent Votes" poll that doesn't appear in the "New Polls" section?

Hmm--I was assuming that "new" and "recent" were relative--that the code was choosing the 5 newest polls and the 5 polls with most recent changes, even if those polls/votes weren't actually all that recent.

If you want five polls in each section, and if there is overlap between them, I guess I figured you'd want to replace the duplicate poll with the next one in line. So, for example, if the list of the five newest polls is identical to the list of the five polls with most recent votes, then I figured that in the bottom section, you'd instead show the polls with the 6th, 7th, 8th, 9th, and 10th most recent votes, so that there were no repeats.

(Or you could do that with the top section instead of the bottom section--leave the bottom section as-is, and change the top section to show the polls that are 6th, 7th, 8th, 9th, and 10th newest.)

File: www/components/poll-sampler.php (hunk 1 above)
Comment (dfabulich) [reply www/components/poll-sampler.php]: Done

**Comment by dfabulich** [comment]
OK, the current version queries for 10 recently voted polls, then removes any of them that were recently created, and takes the top five of those.

**Pull request merged**